packages/frontend/src/components/DashboardTiles/DashboardChartTile.tsx

```diff
@@ -864,6 +829,43 @@ const DashboardChartTileMain: FC<DashboardChartTileMainProps> = (props) => {
         user.data?.ability,
     ]);
 
+    const getDownloadQueryUuid = useCallback(
+        async (csvLimit: number | null) => {
+            const currentQueryUuid =
+                dashboardChartReadyQuery.executeQueryResponse.queryUuid;
+            const currentResultsTotal = rows.length;
+
+            if (
+                dashboardUuid &&
+                (csvLimit === null || csvLimit !== currentResultsTotal)
+            ) {
+                const response = await executeAsyncDashboardChartQuery(
+                    projectUuid!,
+                    {
+                        chartUuid: chart.uuid,
+                        dashboardUuid: dashboardUuid!,
+                        dashboardFilters: appliedDashboardFilters,
+                        dashboardSorts: [],
+                        context: QueryExecutionContext.DASHBOARD,
+                        invalidateCache: false,
+                        ...(csvLimit !== null && { limit: csvLimit }),
+                    },
+                );
+                return response.queryUuid;
```

Comment:
We need to wait for the query results to be ready before triggering the download.  
Check what we have in `executeQueryAndWaitForResults` https://github.com/lightdash/lightdash/blob/092f7e986997c2a6a414f0b3df66ccb60d28f1b6/packages/frontend/src/hooks/useQueryResults.ts#L170